specs/DataSet.rst

```diff
@@ -0,0 +1,253 @@
+    This method returns an opaque subscription identifier.
+
+DataSet.unsubscribe(subid)
+    Removes the indicated subscription.
+    The subid must be the same object that was returned from a DataSet.subscribe call.
+
+Storage
+-------
+
+DataSet.read_from(location, formatter=)
+    Reads a DataSet from persistent store.
+    Location may be a string file system path, a string URL, or some other string that is meaningful to the formatter specified.
+    Formatter is a QCoDeS Formatter object that specifies how data is read and written. 
+    If not provided, the default formatter is used. 
+    The default formatter is currently GNUPlotFormat().
+    This is a static method in the DataSet class.
+    It returns a new DataSet object.
+
+DataSet.read_updates()
+    Updates the DataSet by reading any new results and metadata written since the last read.
+    This method returns a tuple of two Booleans indicating whether or not there were new results and whether or not there was new metadata.
+
+DataSet.write(location, formatter=, overwrite=)
+    Writes the DataSet to persistent store.
+    Location may be a string file system path, a string URL, or some other string that is meaningful to the formatter specified.
+    Formatter is a QCoDeS Formatter object that specifies how data is read and written. 
+    If not provided, the default formatter is used; currently the default is GNUPlotFormat().
+    Overwrite, if true, indicates that any old data found at the specified location should be deleted.
+    Otherwise, it is an error to specify a location that is already in use.
+    This method can be called even if the DataSet is empty, in order to specify the location and format
+
+DataSet.write_updates()
+    Writes new results in the DataSet to persistent store.
+    Depending on the formatter, this may append to an existing stored version or may overwrite the stored version.
+
+DataSet.write_copy(location, formatter=, overwrite=)
+    Writes a separate copy of the DataSet to persistent store.
+    Location may be a string file system path, a string URL, or some other string that is meaningful to the formatter specified.
+    Formatter is a QCoDeS Formatter object that specifies how data is read and written. 
+    If not provided, the formatter for the DataSet is used. 
+    Overwrite, if true, indicates that any old data found at the specified location should be deleted.
+    Otherwise, it is an error to specify a location that is already in use.
+
+Open Issues
+===========
+
+#. Should DataSets automatically write to persistent store periodically, or should the user be required to call write() in order to flush changes ?
+
+At least for now, it seems useful to maintain the current behavior of the DataSet flushing to disk periodically.
+
+#. Should there be a DataSet method similar to add_result that automatically adds a new result by calling the get() method on all parameters that are defined by QCoDeS Parameters?
```

Comment:
No, there should not be such a method. 

That is exactly what the Loop (or analogous function) is intended to do. I think the DataSet should be as simple and modular as possible.